cmd/rancher-desktop-guestagent/main.go

```diff
@@ -159,6 +159,12 @@ func main() {
 	}
 
 	if *enableDocker {
+		if !*enablePrivilegedService {
```

Comment:
This is getting complex enough that having a comment on why we're doing this would be useful.